### ISSUE TITLE
Stabilize new naming conventions

### ISSUE THREAD

**Issue by ken-matsui** [cabinpkg/cabin#1044]
### Idea

- **Files/Directories**: `PascalCase`
- **Types/Classes**: `PascalCase`
- **Variables**: `snake_case`
- **Class (non-struct) Member Variables**: `snake_case_`
- **Functions**: `camelCase`
- **Class (non-struct) Methods**: `camelCase_`

### Reasoning/Explanation

- Variables use `snake_case` since they tend to be shorter than functions.
- Why not the `m` prefix?
    - We need to use `m_` for `snake_case` and `m` for `camelCase`.
    - `m_` and `m` are inconsistent.
    - `m_snake_case` is longer than `snake_case_`.
- Class Members and non-members (and struct members) are distinguished by the `_` suffix, which is similar to [Google's C++ Coding Style](https://google.github.io/styleguide/cppguide.html#Variable_Names).
    - The difference is that we apply this pattern to methods as well.
- Only class members have the `_` suffix because they have methods, and within methods, variables/functions and members are hard to be distinguished. (Structs won't have methods. => this is different from Google's)

### Steps to apply this new policy

1. Apply [Class vs Struct style](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes).
2. Apply the new `Class (non-struct) Methods` naming convention.
3. Apply the new `Class (non-struct) Member Variables` naming convention.
4. Apply the new `Variables` naming convention.

### See also

* https://github.com/poac-dev/poac/blob/bca4bbde7c9eaea9245e53807d49c322583bf187/CONTRIBUTING.md#naming-conventions-under-construction
* https://github.com/poac-dev/poac/pull/1035#discussion_r1843078361
* https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly
* https://llvm.org/docs/Proposals/VariableNames.html

**Comment by ken-matsui** [comment]
This naming convention can be problematic when we want to refer to a member variable via methods:

```
class A {
    int some_thing_;

public:
    int someThing_(); // ok, but depending on the variable name, it might be difficult to read the connection?
                      // Also, somewhat annoying when we want to grep the original variable?
    int getSome_thing_(); // violates camelCase_
    int get_some_thing_(); // violates camelCase_
```

**Comment by badumbatish** [comment]
is this for the cabin codebase itself?

**Comment by ken-matsui** [comment]
Yes, users can control it via .clang-tidy depending on their preferences

**Comment by badumbatish** [comment]
i'm not sure what to make of these warnings, does any of these warnings violate the naming conventions? Or do I have to go to clang-tidy and add the conventions following the filed issue? 

<img width="643" alt="Image" src="https://github.com/user-attachments/assets/a322e515-18ae-4c2e-9880-aea9383a16aa" />

**Comment by ken-matsui** [comment]
IIRC, that’s from system headers. I think I was passing a flag to suppress that useless outputs, but maybe newer clang-tidy doesn’t recognize it anymore? I’ll take a look, but the important thing here is if you have any violations, you should see errors because all warning on this project will be treated as errors: https://github.com/cabinpkg/cabin/blob/7d31d74380a66d2d351f4a749b2c98a6ae0e87ea/.clang-tidy#L3
